### PR TITLE
tuple forwarding and cat fixes

### PR DESCRIPTION
**Pull request by trws**
This is a set of fixes for tuple storage that specifically apply to forwarding and catting tuples.  Mainly reported by @DavidPoliakoff.

### CODE REVIEW

File: include/camp/camp.hpp [hunk 1]
```diff
@@ -151,19 +151,23 @@ namespace test
 }
 #endif
 
+
+/**
+ * @brief Get the index of the first instance of T in L
```
Comment (davidbeckingsale) [inline include/camp/camp.hpp]: Woooo, documentation 😄

**Comment by DavidPoliakoff** [comment]
Thanks for putting this together, be nice to have these merged so I can operate off of RAJA/develop again

File: include/camp/camp.hpp (hunk 1 above)
Comment (trws) [reply include/camp/camp.hpp]: Yup.  Much more is needed, but it seemed like a place to start.  CAMP just got its release today, so I'll probably be taking a pass over all of that in the near future.

**Comment by trws** [comment]
Ironically, given #514, the travis issue here is icc16 not generating an implicit copy constructor that should exist.

**Pull request merged**